### PR TITLE
feat(firestore): add toJSON() and valueOf() to FirestoreTimestamp

### PR DESCRIPTION
**Pull request by pokey**
### Description

For parity with JS SDK, and also to make comparisons work as expected.  Copied from https://github.com/firebase/firebase-js-sdk/blob/7c1c7f182b59e0fc7d175f53e5e2360cdee0ccab/packages/firestore/src/api/timestamp.ts#L162-L182.

I just copied and pasted without testing or modification, so please lmk if this is something you'd be interested in including, and I'd be happy to add a unit test / clean / add type info to proper place

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ ] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [ ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ ] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/firestore/lib/FirestoreTimestamp.js [hunk 1]
```diff
@@ -98,4 +98,26 @@ export default class FirestoreTimestamp {
   toString() {
     return `FirestoreTimestamp(seconds=${this.seconds}, nanoseconds=${this.nanoseconds})`;
   }
+
+  toJSON(): { seconds: number; nanoseconds: number } {
```
Comment (mikehardy) [inline packages/firestore/lib/FirestoreTimestamp.js:102]:
```suggestion
  toJSON(): { seconds: number, nanoseconds: number } {
```

**Review by mikehardy — APPROVED** [review]
@pokey 🏆 ! Sorry for the delay in reviewing this, there was one small lint error I corrected, but otherwise this looked great, I will merge this and do a release in an hour or so

**Comment by davidstott** [comment]
Should FirestoreTimestamp include  MIN_SECONDS for valueOf to work?
Something like: `const MIN_SECONDS = -62135596800;`

**Comment by mikehardy** [comment]
@davidstott great catch! That's what we get for using javascript still instead of typescript 😞 

The relevant lines we are missing are these https://github.com/firebase/firebase-js-sdk/blob/7c1c7f182b59e0fc7d175f53e5e2360cdee0ccab/packages/firestore/src/api/timestamp.ts#L21-L22

```javascript
// The earlist date supported by Firestore timestamps (0001-01-01T00:00:00Z).
const MIN_SECONDS = -62135596800;
```
I'll PR it quickly

**Pull request merged**